Compile bug: ifstream import missing from common.h

### Git commit

$git rev-parse HEAD 3ac3c20c968cc6d485d89382f202ec440767875a

### Operating systems

Linux

### GGML backends

CPU

### Problem description & steps to reproduce

I'm trying to compile Llama.cpp on a Graviton4 with GCC trunk, during the build I get the following error:
```cmake
    859 | std::ifstream fs_open_ifstream(const std::string & fname, std::ios_base::openmode mode);
            |      ^~~~~~~~
/opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:17:1: note: 'std::ifstream' is defined in header '<fstream>'; this is probably fixable by adding '#include <fstream>'
     16 | #include <algorithm>
  +++ |+#include <fstream>
``` 
It seems there is an include header missing from 'common/common.h' which is causing errors for numerous files in 'common/'

### First Bad Commit

 I believe the culprit is this commit: e475fa2b5f9fb50c3d6fc3e7c6fdf1e004465b62

### Compile command

```shell
cmake -S %teamcity.build.checkoutDir% \
  -B . \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_C_FLAGS="$CFLAGS" \
  -DCMAKE_CXX_FLAGS="$CXXFLAGS" \
  -DLLAMA_OPENSSL=ON \
  -DLLAMA_CURL=OFF \
  -DGGML_NATIVE=OFF \
  -DGGML_CPU_AARCH64=ON \
  -DGGML_CPU_ARM_ARCH="%CPU_ARM_ARCH_OPTS%" \
  -DGGML_CPU_KLEIDIAI="%CPU_KLEIDIAI_OPTS%" \
  -DLLAMA_LLGUIDANCE=OFF
cmake --build . --config Release -j --verbose --target llama-batched-bench llama-quantize
```

### Relevant log output

```shell
12:00:11   In file included from /opt/buildAgent/work/c7eaf0c2fd2fe166/common/sampling.h:5,
12:00:11                    from /opt/buildAgent/work/c7eaf0c2fd2fe166/common/llguidance.cpp:1:
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:859:6: error: 'ifstream' in namespace 'std' does not name a type
12:00:11     859 | std::ifstream fs_open_ifstream(const std::string & fname, std::ios_base::openmode mode);
12:00:11         |      ^~~~~~~~
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:17:1: note: 'std::ifstream' is defined in header '<fstream>'; this is probably fixable by adding '#include <fstream>'
12:00:11      16 | #include <algorithm>
12:00:11     +++ |+#include <fstream>
12:00:11      17 |
12:00:11   In file included from /opt/buildAgent/work/c7eaf0c2fd2fe166/common/ngram-map.cpp:1:
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:859:6: error: 'ifstream' in namespace 'std' does not name a type
12:00:11     859 | std::ifstream fs_open_ifstream(const std::string & fname, std::ios_base::openmode mode);
12:00:11         |      ^~~~~~~~
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:17:1: note: 'std::ifstream' is defined in header '<fstream>'; this is probably fixable by adding '#include <fstream>'
12:00:11      16 | #include <algorithm>
12:00:11     +++ |+#include <fstream>
12:00:11      17 |
12:00:11   In file included from /opt/buildAgent/work/c7eaf0c2fd2fe166/common/speculative.h:4,
12:00:11                    from /opt/buildAgent/work/c7eaf0c2fd2fe166/common/speculative.cpp:1:
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:859:6: error: 'ifstream' in namespace 'std' does not name a type
12:00:11     859 | std::ifstream fs_open_ifstream(const std::string & fname, std::ios_base::openmode mode);
12:00:11         |      ^~~~~~~~
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:17:1: note: 'std::ifstream' is defined in header '<fstream>'; this is probably fixable by adding '#include <fstream>'
12:00:11      16 | #include <algorithm>
12:00:11     +++ |+#include <fstream>
12:00:11      17 |
12:00:11   In file included from /opt/buildAgent/work/c7eaf0c2fd2fe166/common/reasoning-budget.cpp:2:
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:859:6: error: 'ifstream' in namespace 'std' does not name a type
12:00:11     859 | std::ifstream fs_open_ifstream(const std::string & fname, std::ios_base::openmode mode);
12:00:11         |      ^~~~~~~~
12:00:11   /opt/buildAgent/work/c7eaf0c2fd2fe166/common/common.h:17:1: note: 'std::ifstream' is defined in header '<fstream>'; this is probably fixable by adding '#include <fstream>'
12:00:11      16 | #include <algorithm>
12:00:11     +++ |+#include <fstream>
12:00:11      17 |
```

--- a/common/common.h
+++ b/common/common.h
@@ -14,6 +14,7 @@
 #include <vector>
 #include <map>
 #include <algorithm>
+#include <fstream>
 
 #if defined(_WIN32) && !defined(_WIN32_WINNT)
 #define _WIN32_WINNT 0x0A00

